File: client/api/vitals.ts
import { showVitals, hideVitals } from "../modules/vitals"

export const getFxPlayerData = () => {
  const QBCore = exports['qb-core'].GetCoreObject()
  const PlayerData = QBCore.Functions.GetPlayerData()
  return PlayerData
}

onNet('QBCore:Client:OnPlayerLoaded', () => {
  showVitals()
})

onNet('QBCore:Client:OnPlayerUnload', () => {
  hideVitals()
})

File: client/api/vitals.lua
local vitals<const> = require 'client/modules/vitals'

local function getFxPlayerData()
  local QBCore<const> = exports['qb-core']:GetCoreObject()
  return QBCore.Functions.GetPlayerData()
end

RegisterNetEvent('QBCore:Client:OnPlayerLoaded', function()
  vitals.showVitals()
end)

RegisterNetEvent('QBCore:Client:OnPlayerUnload', function()
  vitals.hideVitals()
end)

return {
  getFxPlayerData = getFxPlayerData
}